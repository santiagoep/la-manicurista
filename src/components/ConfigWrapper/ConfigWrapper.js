import React, { useEffect } from "react";
import { useSelector } from "react-redux";
import { ThemeProvider } from "styled-components";

import i18n from "@config/translations";
import themes from "@assets/styles/themes";
import BaseStyles from "@assets/styles/BaseStyles";
import childrenProps from "@utils/common-propTypes";

const ConfigWrapper = ({ children }) => {
  const { theme, fontSize, language } = useSelector(({ config }) => config);
  const selectedTheme = themes.find(({ name }) => name === theme);

  useEffect(() => {
    i18n.changeLanguage(language);
  }, [language]);

  return (
    <ThemeProvider theme={selectedTheme}>
      <BaseStyles fontSize={fontSize} />
      {children}
    </ThemeProvider>
  );
};

ConfigWrapper.propTypes = {
  children: childrenProps.isRequired,
};

export default ConfigWrapper;
