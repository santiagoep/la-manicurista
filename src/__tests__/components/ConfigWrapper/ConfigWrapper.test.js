import React from "react";

import render from "../../render";
import ConfigWrapper from "@components/ConfigWrapper/ConfigWrapper";

describe("<ConfigWrapper />", () => {
  it("should render snapshot", () => {
    const { renderer } = render({ theme: true, redux: true });
    const component = renderer(
      <ConfigWrapper>
        <h1>Test</h1>
      </ConfigWrapper>
    );
    expect(component).toMatchSnapshot();
  });
});
