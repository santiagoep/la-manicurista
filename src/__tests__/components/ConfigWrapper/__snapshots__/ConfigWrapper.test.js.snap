// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigWrapper /> should render snapshot 1`] = `
<h1>
  Test
</h1>
`;
